medusa/providers/nzb/binsearch.py

```diff
@@ -6,120 +6,236 @@
-        """Initialize the class."""
-        kwargs.pop('search_params', None)  # does not use _get_rss_data so strip param from kwargs...
-        search_params = None  # ...and pass None instead
-        tv.Cache.__init__(self, provider_obj, search_params=search_params, **kwargs)
+        def process_column_header(td):
+            return td.get_text(strip=True).lower()
+
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            table = html.find('table', class_='xMenuT')
+            rows = table('tr') if table else []
+            row_offset = 1
+            if not rows or not len(rows) - row_offset:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            headers = rows[0]('th')
+            # 0, 1, subject, poster, group, age
+            labels = [process_column_header(header) or idx
+                      for idx, header in enumerate(headers)]
+
+            # Skip column headers
+            rows = rows[row_offset:]
+            for row in rows:
+                try:
+                    col = dict(zip(labels, row('td')))
+                    nzb_id_input = col[0 if mode == 'RSS' else 1].find('input')
+                    if not nzb_id_input:
+                        continue
+                    nzb_id = nzb_id_input['name']
+                    # Try and get the the article subject from the weird binsearch format
+                    if mode == 'RSS':
+                        title_field = col['subject']
+                        title = BinSearchProvider.title_regex_rss.search(title_field.text).group(1)
+                    else:
+                        title_field = col['subject']
+                        title = BinSearchProvider.title_regex.search(title_field.text).group(1)
+                    for extension in ('.nfo', '.par2', '.rar', '.zip', '.nzb'):
+                        # Strip extensions that aren't part of the file name
+                        title = title.rstrip(extension)
+                except AttributeError:
+                    log.debug('Parsing rows, that may not always have usefull info. Skipping to next.')
+                    continue
+                if not all([title, nzb_id]):
+                    continue
+                # Obtain the size from the 'description'
+                size_field = BinSearchProvider.size_regex.search(col['subject'].text)
+                if size_field:
+                    size_field = size_field.group(1)
+                size = convert_size(size_field, sep='\xa0') or -1
+                print(size)
+                size = int(size)
+
+                query = query or title
+
+                download_url = 'https://www.binsearch.info/fcgi/nzb.fcgi?q={query}&max=100&adv_age=1100&server='.format(
+                    query=query
+                )
+
+                download_url = '{download_url}|nzb_id={nzb_id}'.format(download_url=download_url, nzb_id=nzb_id)
```

Comment:
@medariox i'm now storing the url and the id (which is needed for the post) using a pipe. We could also add a field to the cache db's for this, but as its only one provider that needs this, I feel this is good enough for now.
What do you think?

Reply:
Agreed, I think it's alright if it remains an exception (1-2 providers).